fv/connectivity/conncheck.go

```diff
@@ -741,3 +743,139 @@ func (cc ConnConfig) GetTestMessageSequence(msg string) (int, error) {
 func IsMessagePartOfStream(msg string) bool {
 	return strings.HasPrefix(strings.TrimSpace(msg), ConnectionTypeStream)
 }
+
+// Runtime abstracts *containers.Container to avoid import loops
+type Runtime interface {
+	EnsureBinary(name string)
+	ExecMayFail(cmd ...string) error
+}
+
+type PermanentConnection struct {
```

Comment:
Nice to have this extracted out, thanks!

Wonder if we should change the name to `PersistentConnection`.  It's a subtle distinction, but I think that's more correct.

Reply:
yeah, will rename it, tbh I was first grepping for persistent ;-)